docs/src/content/docs/reference/configuration.mdx

```diff
@@ -65,6 +65,33 @@ type LogoConfig = { alt?: string; replacesTitle?: boolean } & (
 
 Configure the table of contents shown on the right of each page. By default, `<h2>` and `<h3>` headings will be included in this table of contents.
 
+### `expressiveCode`
+
+**type:** `StarlightExpressiveCodeOptions | boolean`
+
+Starlight uses [Expressive Code](https://github.com/expressive-code/expressive-code/tree/main/packages/astro-expressive-code) to render code blocks on your site. The integration is enabled by default and can be disabled by passing `false` to this option.
+
+You can customize your code blocks by passing an object with any of the following properties:
+
+- `themes` — an array of themes that should be used to render code blocks. The following item types are supported in this array:
+  - any theme name bundled with Shiki (e.g. `dracula`)
+  - any theme object compatible with VS Code or Shiki (e.g. imported from an NPM theme package)
+  - any ExpressiveCodeTheme instance (e.g. using `ExpressiveCodeTheme.fromJSONString(...)` to load a custom JSON/JSONC theme file yourself)
+
+  Defaults to the dark and light variants of the "Night Owl" theme by Sarah Drasner, which are also exported from `@astrojs/starlight/expressive-code` as `getStarlightDefaultThemes`.
+
+  **Note**: If you pass at least one dark and one light theme to this option, Starlight will automatically ensure that the displayed theme matches the current dark mode switch state. See the `useStarlightDarkModeSwitch` option to configure this behavior. If you turn it off, you will need to add your own CSS code to the site to allow manual switching between themes.
+
+- `useStarlightDarkModeSwitch` — determines if CSS code should be added that automatically displays dark or light code blocks depending on Starlight's dark mode switch state.
+
+  Defaults to `true` if the `themes` option is not set or contains at least one dark and one light theme, and `false` otherwise.
```

Comment:
Wondering what the use case for this option is? I think it’s potentially confusing because `false` does not say “I want always dark” or “I want always light”, so I guess it only really works well when you also set `themes` to a single theme?

My instinct would be to suggest removing this in favour of respecting `themes`: single theme persists across dark and light modes, two themes toggles with the dark mode switch. Unless we have a clear use case this solves for?

Reply:
If you leave this option undefined, the default behavior is meant to be just like you described and automatically respect what you configured in `themes`.

The use case I was thinking of when adding this option was to allow people to provide their own logic for code block theme switching (e.g. by adding custom UI to select a preferred syntax highlighting theme), instead of the code blocks reacting automatically to the Starlight theme switcher.

I can understand if you'd prefer leaving this option out until there are actual users requesting it to be added. Just let me know and I can remove it! :)